### PR TITLE
dev-lang/crystal: version bump to 0.16.0

### PR DESCRIPTION
**Pull request by longlene**

### CODE REVIEW

**Comment by wraeth** [comment]
Hi!

This seems to include a bump and clean of both `dev-lang/crystal` and `dev-util/shards`, but the commits are shared. The changes to each package should be done in their own commits to preserve atomicity - one commit bumping crystal, one bumping shards, one cleaning old version from crystal, and again one for shards.

Otherwise, the bumped ebuilds look good at first glance, so I will have a closer look shortly and run some quick testing. Would you be able to restructure the commits and force-push to update this PR?

**Comment by wraeth** [comment]
@gentoo/proxy-maint

**Comment by longlene** [comment]
@wraeth Maybe I can rebase the commit if it is not too late.:-)

**Comment by wraeth** [comment]
I haven't pushed this yet, so you can restructure and rebase as appropriate.

I seem to be getting build failures, however, though I'm not sure if it's caused by my testing environment or something else. I will investigate further and see what I can find, but in the meantime: have you built this successfully yourself?

**Comment by longlene** [comment]
@wraeth Yes,I've built it. dev-lang/crystal need <llvm-3.7

**Comment by wraeth** [comment]
Thanks for rebasing this.

I'm still getting build failures in my testing environment, so I will try building this on my live host to see if there is an issue there. The environment I was testing in was using `llvm-3.6.2`, so I don't believe that was the issue. The lack of output from the crystal build does make this somewhat harder, though. :smile:

One thing of note, however, is that I noticed you are setting `THREADS=4` in the ebuild. If I understand correctly, this would mean that four build processes would be used, regardless of the capabilities of the building system (for example, a two-core laptop would still be trying to use four threads). If that's the case, then this should probably either search the `${MAKEOPTS}` variable and extract the number of jobs from that or inherit [`multiprocessing.eclass`](https://devmanual.gentoo.org/eclass-reference/multiprocessing.eclass/) and use the provided function for getting the number of jobs.

**Comment by longlene** [comment]
@wraeth I can't reproduce the build failure, multi-threaded building may result in a race condition. It may be better to remove the "THREADS=4" line?

**Comment by monsieurp** [comment]
what's the status here? can we get on with this PR?

**Comment by longlene** [comment]
@wraeth Could you please reproduce the compile issue and post the error message? Thanks:-)

File: dev-lang/crystal/crystal-9999.ebuild [hunk 1]
```diff
@@ -1,24 +1,35 @@
...
 # Distributed under the terms of the GNU General Public License v2
 # $Id$
 
-EAPI=5
+EAPI=6
 
 inherit multilib
 
-BV=${PV}-1
+MY_BSV=0.17.4
```
Comment (mgorny) [inline dev-lang/crystal/crystal-9999.ebuild]: Does it really make sense to introduce this variable, if you're only using it to build BV?

**Comment by mgorny** [comment]
The commit wrt parallel compiling lacks prefix in commit message. It would also probably make sense to rebase and remove the intermediate bump.

File: dev-lang/crystal/crystal-9999.ebuild (hunk 1 above)
Comment (longlene) [inline dev-lang/crystal/crystal-9999.ebuild]: Because the crystal can't bootstrap itself, I use this binary package for compiling.

**Comment by mgorny** [comment]
Ping. @wraeth, are you still handling this? @longlene, I see that it has conflicts now, could you look at it?

**Comment by longlene** [comment]
@mgorny As someone has bump the version to 0.20.0, may this pr is useless now.

**Pull request closed without merging**